compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -721,6 +721,16 @@ impl<'a, G: EmissionGuarantee> Diag<'a, G> {
         self
     }
 
+    #[rustc_lint_diagnostics]
```

Comment:
Can you inline this into the one caller that it's used? Also, this could be a `Subdiagnostic`, perhaps? I'm not sure if `highlighted_note` handles translatable string parts...

Reply:
Yes, I agree this should probably be `Subdiagnostic`, however the  `highlighted_note` is accepting only `StringPart` and not the `impl Into<SubdiagMessage>` (which is the translatable string part right?). Should address the `highlighted_note` so it is accepting the `SubdiagMessage` in this PR? 
Also, we can do it without ~colors~highlighting for now.

Reply:
Nah, it's fine.